impl_unix.go

```diff
@@ -301,7 +312,14 @@ func listenUDP(netw, addr string) (c net.Conn, err error) {
 }
 
 // this is close to the connect() function inside stdlib/net
-func connect(fd int, ra syscall.Sockaddr, deadline time.Time) error {
+func connect(ctx context.Context, fd int, ra syscall.Sockaddr, deadline time.Time) error {
+	// NOTE: We currently only use the context to pass in a deadline.
+	// Canceling an epoll_wait call is hard.
```

Comment:
This is a bit cheaty but it might work, adding a UDP socket to epoll_wait group and using it as wakeup call. It might work with just one socket and SO_REUSEPORT but fifo probably would work as well.